test/data-simplest.jl

```diff
@@ -0,0 +1,120 @@
+        :investment_limit_storage_energy => [1.0, 2.0],
+    ),
+    "asset_milestone" => DataFrame(
+        :asset => ["some_producer", "some_consumer"],
+        :milestone_year => [2030, 2030], # TODO: Should be required?
+        :min_energy_timeframe_partition => [1.0, 1.0],
+        :max_energy_timeframe_partition => [2.0, 2.0],
+    ),
+
+    # Basic flow data
+    "flow" => DataFrame(
+        :from_asset => ["some_producer"],
+        :to_asset => ["some_consumer"],
+        :carrier => ["whatever"], # TODO: Why do we need this?
+    ),
+    "flow_both" => DataFrame(
+        :from_asset => ["some_producer"],
+        :to_asset => ["some_consumer"],
+        :commission_year => [2030], # TODO: Should be required?
+        :milestone_year => [2030], # TODO: Should be required?
+    ),
+    "flow_commission" => DataFrame(
+        :from_asset => ["some_producer"],
+        :to_asset => ["some_consumer"],
+        :commission_year => [2030], # TODO: Should be required?
+        :fixed_cost => [1.0],
+        :investment_cost => [1.0],
+        :investment_limit => [1.0],
+    ),
+    "flow_milestone" => DataFrame(
+        :from_asset => ["some_producer"],
+        :to_asset => ["some_consumer"],
+        :milestone_year => [2030], # TODO: Should be required?
+    ),
+
+    # Basic time information
+    "year_data" => DataFrame(
+        :year => [2030], # TODO: Should be required?
+        :length => [8760],
+        :is_milestone => [true],
+    ),
+    "rep_periods_data" => DataFrame(
+        :year => [2030, 2030], # TODO: Should be required?
+        :rep_period => [1, 2], # TODO: Should be required?
+        :num_timesteps => [24, 24], # TODO: Should be required?
+    ),
+    "timeframe_data" => DataFrame(
+        :year => 2030, # TODO: Should be required?
+        :period => 1:365,
+        :num_timesteps => 24, # TODO: Should be required?
+    ),
+    "rep_periods_mapping" => DataFrame(
+        :year => 2030, # TODO: Should be required?
+        :period => 1:365, # TODO: Should be required?
+        :rep_period => mod1.(1:365, 2), # TODO: Should be required?
+    ),
+
+    # Profiles
+    "assets_profiles" => DataFrame(
+        "asset" => ["some_producer"],
+        "commission_year" => [2030], # TODO: Should be required
+        "profile_name" => ["some_producer_profile"],
+        "profile_type" => ["availability"],
+    ),
+    "flows_profiles" => DataFrame(
+        "from_asset" => ["some_producer"],
+        "to_asset" => ["some_consumer"],
+        "year" => [2030], # TODO: Should be required
+        "profile_name" => ["some_flow_profile"],
+        "profile_type" => ["availability"],
+    ),
+    "profiles_rep_periods" => DataFrame(
+        "profile_name" => "some_producer_profile",
+        "year" => 2030, # TODO: Should be required
+        "rep_period" => [fill(1, 24); fill(2, 24)], # TODO: should be required
+        "timestep" => [1:24; 1:24], # TODO: should be required
+        "value" => ones(48),
+    ),
+)
```

Comment:
@datejada @gnawin I'm creating the simplest data that has no defaults. Can you check these? Notice I added some "TODO" to check whether the listed data should have defaults or not. 
I am assuming that this data is for TEM after clustering, so rep_period things are included as required.
For some, I guess it's convenient for the user to have them friendly included, but for TEM I don't think that should be the case (like the years).